### PR TITLE
Improve description of "coqc -Q"

### PR DESCRIPTION
**Pull request by jfehrle**
There was no description of how to compile a single file with `coqc` and load it into a Coq session--an elementary use case that beginners will run into frequently, as I did.

BTW, I notice that `.. coqdoc::` directives generate an extra space above the output.  This seems to be from the CSS but I haven't figured it out:

image

### CODE REVIEW

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (ejgallego) [inline doc/sphinx/practical-tools/coq-commands.rst:65]: Please don't use this command in example code, it is not recommended to setup your workspace with vernacular commands, but instead using the corresponding flags.

Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: Meaning in _CoqProject or on the command if you must?  At the moment, both require restarting coqtop/coqide.  Perhaps coqide could be modified so it doesn't need to be restarted.

I agree `Add LoadPath` is not great, but it doesn't require restarting CoqIDE, which is a bit disruptive.  So maybe reasonable for quick experiments.

Comment (SkySkimmer) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: For coqide putting the flags in coqproject then closing/reopening the .v file should work.

Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: Less bad, thanks, but it should be more seamless in CoqIDE.  Opening a new scratch buffer doesn't reload the info, which I expected.

Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: Looks like CoqIDE lets you set the name of the project file in preferences.  `coqtop` doesn't seem to apply the default name, nor do I see a command line option for providing the name.  That's a bug, too, even if normal users won't use coqtop.

Comment (SkySkimmer) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: >That's a bug, too, even if normal users won't use coqtop.

What does that mean? coqtop is kept for proffgeneral and that's about it.

Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:65]: We use it in the development process and tests, too.

**Comment by jfehrle** [comment]
Updated.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coq-commands.rst:83]: `coqtop` and `coqc` only have `-R` / `-Q` options that you must pass manually. The idea is that users are never supposed to call these commands directly. A user should use an editor, which should load `_CoqProject` and `coq_makefile` or `dune` to build.

File: doc/sphinx/language/core/modules.rst
Comment (Zimmi48) [inline doc/sphinx/language/core/modules.rst:996]:
```suggestion
short names ``foo.Bar.File``, ``Bar.File`` and ``File``. If several files with
```

File: doc/sphinx/practical-tools/utilities.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/utilities.rst:602]: Yes, `COQPATH` works and is heavily relied on in the Nix setup. It is documented in the modules chapter, in one of the places you edited in this PR.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:50]: There is no restriction on the validity of the directory which contains a `.v` file:

```
$ coqc test\ with\ space/test.v 
```
works.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:55]: :+1:

```suggestion
We recommend specifying a logical directory name (which is also the module name)
with the `-R` or the `-Q` options.
Generally we recommend using provided utilities such as `coq_makefile` or `dune` to build Coq projects.
See :ref:`building_coq_project`.
```

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:74]: What is `empty_dir` supposed to represent? Wouldn't it be better to just recommend that the `_CoqProject` file be always located at the root of a project and that CoqIDE (or another compatible editor) be always run from the root of the project as well?

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:75]: Note that listing files is only useful for `coq_makefile`. Only the options on the first lines matter for editors like CoqIDE.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:80]:
```suggestion
   it non-portable. Therefore, this is discouraged.
```

**Review by Zimmi48 — APPROVED** [review]
Thanks for improving the doc and adding an example. This is one of the things that are clearly lacking in the current doc, and a frequent source of troubles for beginners.

File: doc/sphinx/practical-tools/utilities.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/utilities.rst:602]: The doc for that is inadequate: it is not complete nor easy to understand nor is there an example for this elementary bit of setup.  Hence my comment.  The format is the same as $PATH (colon separated on Linux, semicolon separated on Windows)?  And everything gets added to the Coq module?  And I have no idea about XDG* or how they would be used--are they platform specific?  Would also be nice to give some guidance about when to use COQPATH versus other mechanisms.

> the directories listed in the $COQPATH, ${XDG_DATA_HOME}/coq/ and ${XDG_DATA_DIRS}/coq/ environment variables (see XDG base directory specification) with the same physical-to-logical translation and with an empty logical prefix.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:50]: This is not new text.  How does such a name get mapped to a module name?  I see such names as something to avoid even if they're permitted.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:55]: `coq_makefile` creates a makefile that you then run with `make`, right?

Made it: "Generally we recommend using utilities such as `make` or `dune` to build Coq projects,
using `coq_makefile` to generate the dependencies."

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (ejgallego) [reply doc/sphinx/practical-tools/coq-commands.rst:80]: It could also be the case that these commands get removed in the future, as they commit to a library organization model which may not hold. Personally, I wouldn't recommend them under any circumstance.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:74]: I removed empty_dir, that was my local test.

Having to start CoqIDE in a specific directory is inconvenient if you start it from an icon on your desktop.  Icons don't allow for changing the directory.  CoqIDE should let you pick the project within its GUI.  I think there are probably some small code tweaks we could make that would make for a better user experience.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:75]: I replaced with a vertical ellipsis (a colon).

**Comment by jfehrle** [comment]
Updated.  Let's take what's good here and not turn this into a larger project.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst]: We document Coq as it is, not as we want it to become.  Only by mentioning and explaining this feature can we say not to use it.  Feel free to start a discussion on removing it--though I think we should review how to use the various compilation options first and perhaps tweak them.

File: doc/sphinx/practical-tools/utilities.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/utilities.rst]: I guess that the format is the same as `$PATH`. I'm not sure, but it's likely. A lot of programming language have a similar `$XXXPATH` environment variable.

>And everything gets added to the Coq module?

I'm not sure how it works exactly but basically it extends the places that Coq will treat like the `user-contrib/` directory in the Coq installation.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coq-commands.rst:50]: You extended the existing text. The previous test only talked about the filename (i.e. last component). I don't know exactly how things work, so I cannot answer your question precisely.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coq-commands.rst]: > `coq_makefile` creates a makefile that you then run with `make`, right?

Yes.
 
> Made it: "Generally we recommend using utilities such as `make` or `dune` to build Coq projects,
> using `coq_makefile` to generate the dependencies."

Maybe tweak it further to:

"Generally we recommend using utilities such as `make` (using `coq_makefile` to generate the `Makefile`) or `dune` to build Coq projects"
> ."

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coq-commands.rst]: I agree.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [reply doc/sphinx/practical-tools/coq-commands.rst]: Is this not going to be interpreted as a literal colon by some users?

**Comment by jfehrle** [comment]
Updated.  Any other changes needed?

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:75]: It's still not clear to me why you use `../mod1` here instead of `.` as in:
```suggestion
         -R . Mod1
```

Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst]: I did something better than your suggestion.

Updated.

**Review by Zimmi48 — APPROVED** [review]
Thanks, can you squash?

**Comment by jfehrle** [comment]
Done.

**Comment by Zimmi48** [comment]
No need to wait for CI, the changes are exactly the same.

@coqbot merge now

**Pull request merged**
